draft-ietf-jsonpath-base.md

```diff
@@ -825,7 +825,7 @@ An index selector `<index>` matches at most one array element value.
 ~~~~ abnf
 index-selector = int                             ; decimal integer
 
-int            = ["-"] ( "0" / (DIGIT1 *DIGIT) ) ; -  optional
+int            = "0" / (["-"] DIGIT1 *DIGIT)     ; -  optional
```

Comment:
I think this deserves another note (#3) below along these lines:

3. The syntax does *not* allow `-0` (which is a valid JSON number).

(Editorial observation: in the base text, it's a little odd that we have the notes below, but we don't have similar notes in the array slice selector syntax section. No need to change this in the current PR, but it wouldn't harm.)

Reply:
I think I'd go the inverse direction and delete the "; - optional" comment -- the ABNF really should speak for itself.

Note that JSON does not have integers, and we do allow -0 in JSON numbers.
I'm having a hard time imagining anyone being too surprised about not being able to use -0 as an integer.

Reply:
Interesting. I'm comfortable with the policy of letting the ABNF speak for itself, but I know of at least one reader of the spec who said that he didn't pay much attention to the ABNF. Anyway, let's not switch policy in this PR. Approving.